src/Playwright/Transport/Channels/PageChannel.cs

```diff
@@ -98,7 +98,7 @@ internal override void OnMessage(string method, JsonElement? serverParams)
                 WebSocket?.Invoke(this, serverParams?.GetProperty("webSocket").ToObject<WebSocketChannel>(Connection.DefaultJsonSerializerOptions).Object);
                 break;
             case "download":
-                Download?.Invoke(this, serverParams?.ToObject<PageDownloadEvent>(Connection.DefaultJsonSerializerOptions));
+                Download?.Invoke(this, (serverParams.Value.GetProperty("url").ToObject<string>(Connection.DefaultJsonSerializerOptions), serverParams.Value.GetProperty("suggestedFilename").ToObject<string>(Connection.DefaultJsonSerializerOptions), serverParams.Value.GetProperty("artifact").ToObject<ArtifactChannel>(Connection.DefaultJsonSerializerOptions)));
```

Comment:
The new one is more verbose, not sure how better it is compared to the previous version.